### PR TITLE
Aprillion improvements

### PR DESCRIPTION
**Pull request by kentcdodds**
Supersedes #26 

cc @Aprillion

### CODE REVIEW

**Comment by kentcdodds** [comment]
@all-contributors please add @Aprillion for code and tests

File: src/react-app.js [hunk 1]
```diff
@@ -386,56 +410,22 @@ function renderReactApp({
...
-                        />
-                      ) : null}
-                    </Sandbox>
+                      files={exercise}
+                    />
                   </TabPanel>
                   <TabPanel>
-                    <Sandbox
+                    <FileTabs
+                      key={exerciseNumber}
```
Comment (Aprillion) [inline src/react-app.js:424]: I guess I forgot to explain the purpose of these `key`s - if not obvious, they replace this deleted workaround:

```js
    // handle this case:
    // 1. Select extra credit tab
    // 2. navigate to exercise with no extra credit
    // we want to set the tab index to 0 so they're not looking at nothing.
    React.useEffect(() => {
      if (!extraCredit.length && tabIndex === 2) setTabIndex(0)
    }, [extraCredit.length, tabIndex])
```

If you think someone might be tempted to remove them as "not needed", please reintroduce the explanation comment ;)

Comment (kentcdodds) [reply src/react-app.js:424]: I'll keep that in mind. Thanks!

**Pull request merged**